src/models/__tests__/profile.test.ts

```diff
@@ -275,6 +319,51 @@ describe("createProfile", () => {
       expect(result.left.kind).toEqual("COSMOS_EMPTY_RESPONSE");
     }
   });
+
+  it.each`
+    case                                                              | pushNotificationsContentType | expectedPushNotificationsContentType
+    ${"existing profile with pushNotificationsContentType empty"}     | ${undefined}                 | ${"UNSET"}
+    ${"existing profile with pushNotificationsContentType UNSET"}     | ${"UNSET"}                   | ${"UNSET"}
+    ${"existing profile with pushNotificationsContentType FULL"}      | ${"FULL"}                    | ${"FULL"}
+    ${"existing profile with pushNotificationsContentType ANONYMOUS"} | ${"ANONYMOUS"}               | ${"ANONYMOUS"}
+  `(
+    "should create a profile with pushNotificationsContentType=$expectedPushNotificationsContentType when passing $pushNotificationsContentType",
+    async ({
+      pushNotificationsContentType,
+      expectedPushNotificationsContentType
+    }) => {
+      const containerMock = ({
+        items: {
+          create: jest.fn().mockReturnValue(
+            Promise.resolve({
+              resource: {
+                ...aRetrievedProfile,
+                pushNotificationsContentType
+              }
+            })
+          )
+        }
+      } as unknown) as Container;
+
+      const model = new ProfileModel(containerMock);
+
+      const newProfileWithNotificationsContentType: NewProfile = {
+        ...newProfile,
+        pushNotificationsContentType
+      };
+      const result = await model.create(
+        newProfileWithNotificationsContentType
+      )();
+
```

Comment:
Check that we are calling the `create` method with the right parameters, if we don't we are testing the mock result only.
```suggestion
      expect(containerMock.items.create).toBeCalledTimes(1);
      expect(containerMock.items.create).toHaveBeenCalledWith(
        expect.objectContaining({
          kind: newProfileWithNotificationsContentType.kind,
          fiscalCode: newProfileWithNotificationsContentType.fiscalCode,
          pushNotificationsContentType: expectedPushNotificationsContentType,
          id: generateVersionedModelId<Profile, "fiscalCode">(
            newProfileWithNotificationsContentType.fiscalCode,
            0 as NonNegativeInteger
          ),
          version: 0
        }),
        expect.anything()
      );
```